libraries/Microsoft.Bot.Connector/Authentication/ConnectorFactoryImpl.cs

```diff
@@ -36,7 +36,11 @@ public override async Task<IConnectorClient> CreateAsync(string serviceUrl, stri
             var credentials = await _credentialFactory.CreateCredentialsAsync(_appId, audience ?? _toChannelFromBotOAuthScope, _loginEndpoint, _validateAuthority, cancellationToken).ConfigureAwait(false);
 
             // A new connector client for making calls against this serviceUrl using credentials derived from the current appId and the specified audience.
-            return new ConnectorClient(new Uri(serviceUrl), credentials, _httpClient, disposeHttpClient: _httpClient == null);
+#pragma warning disable CA2000 // Dispose objects before losing scope
+            var httpClient = _httpClientFactory?.CreateClient() ?? new HttpClient();
+            ConnectorClient.AddDefaultRequestHeaders(httpClient);
+            return new ConnectorClient(new Uri(serviceUrl), credentials, httpClient, true);
```

Comment:
So just to confirm: you're not disposing the httpClient because you are passing disposeHttpClient to the ConnectorClient base class, explicitly granting ownership of that object to the ConnectorClient. Is that correct? If yes, might be worth a code comment.

Reply:
I'm passing true, I read that as meaning "call dispose" is that correct?

so the only time this will actually matter is if you don't use the factory (which you should!) in which case we will create and then dispose the (relatively expensive) HttpClient.

So its correct either way - because calling Dispose or failing to call Dispose on a HttpClientFactory produced HttpClient is good either way. i.e. its harmless